corehq/apps/users/models.py

```diff
@@ -1813,7 +1813,7 @@ def get_owner_ids(self, domain=None):
         owner_ids.extend([g._id for g in self.get_case_sharing_groups()])
         return owner_ids
 
-    def unretire(self):
+    def unretire(self, unretired_by, unretired_via=None):
```

Comment:
The difference between these two new parameters is not obvious to me. Why is one required and one optional? Would be nice if those details were noted in the docstring.

Reply:
also since there is only one way to do this it seems unnecessary

Reply:
i am happy to revisit this, just sharing the reason behind this was to avoid passing two `None` values where these are not needed, specifically in tests, which I found were many when I added for delete/retire in
https://github.com/dimagi/commcare-hq/blob/f776b98d49f7fcd54f4e8ce30e58bedb429cdc27/corehq/apps/users/models.py#L1847 but are less here.

I think I introduced a confusion because i made both arguments mandatory when i added these for create, first of such audit tracking, in 
https://github.com/dimagi/commcare-hq/blob/f776b98d49f7fcd54f4e8ce30e58bedb429cdc27/corehq/apps/users/models.py#L2388 where i learnt the weight to adding two `None` in tests.

Just adding that, It is expected that when a method passes "*_by", it would also pass "*_by" along with it. And if not needed, currently for delete and unretire, one can just pass `None` for "*_by", but that is only limited to tests.

Reply:
(will take this change, as concluded, as a followup)